app/controllers/to_do_lists_controller.rb

```diff
@@ -1,10 +1,11 @@
 class ToDoListsController < ApplicationController
 
   def index
-    lists = Finder.new(cookies[:filter]).perform
-    @lists_presentation = ListsPresentation.new lists
+    @lists_per_page = Finder.new(cookies[:filter]).perform
+    @lists_per_page = @lists_per_page.paginate(page: params[:page])
```

Comment:
@vlasiak, I'm not fan of reassigning of the variable. I suggest to use another local variable for this or better to rewrite/reformat the code.